src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -95,12 +109,16 @@ impl {{ target_kind_name }} {
             {% endfor %}
             writer.step_out()?;
         {% elif abstract_data_type is object and abstract_data_type is containing("Sequence")  %}
-            writer.step_in(IonType::List)?;
+            {% if abstract_data_type["Sequence"].sequence_type == "list" %}
+                writer.step_in(IonType::List)?;
+            {% else %}
+                writer.step_in(IonType::SExp)?;
+            {% endif %}
```

Comment:
(🗺️ PR tour) Adds exception statements for when `sequence_type`(list or sexp) doesn't match while reading Ion data.